src/legacy/core_plugins/visualizations/public/np_ready/public/legacy/build_pipeline.ts

```diff
@@ -86,12 +90,16 @@ const getSchemas = (
   vis: Vis,
   opts: {
     timeRange?: any;
+    timefilter: TimefilterContract;
   }
 ): Schemas => {
   const { timeRange } = opts;
   const createSchemaConfig = (accessor: number, agg: IAggConfig): SchemaConfig => {
     if (isDateHistogramBucketAggConfig(agg)) {
       agg.params.timeRange = timeRange;
+      // TODO: This is only place this is used outside the `data` plugin.
+      // We should find a way to get rid of it so it can stay internal to `data`.
+      updateTimeBuckets(agg, opts.timefilter);
```

Comment:
replace by agg.buckets.setBounds() to not expose the utility externally ? i am also fine doing this in a followup